metricingester/sinkflusher.go

```diff
@@ -0,0 +1,93 @@
+package metricingester
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/stripe/veneur/trace/metrics"
+
+	"github.com/sirupsen/logrus"
+	"github.com/stripe/veneur/samplers"
+	"github.com/stripe/veneur/ssf"
+	"github.com/stripe/veneur/trace"
+)
+
+type sinkFlusher struct {
+	aggregates  samplers.HistogramAggregates
+	percentiles []float64
+	sinks       []Sink
+	log         *logrus.Logger
+	tc          *trace.Client
+}
+
+type Sink interface {
+	Name() string
+	Flush(context.Context, []samplers.InterMetric) error
+}
+
+func (s sinkFlusher) Flush(ctx context.Context, envelope samplerEnvelope) {
+	logger := traceLogger(s.log, ctx)
+	ms := make([]samplers.InterMetric, 0, countMetrics(envelope))
+	// get ms from envelope
+	for _, sampler := range envelope.counters {
+		ms = append(ms, sampler.Flush(time.Second)...)
+	}
+	for _, sampler := range envelope.sets {
+		ms = append(ms, sampler.Flush()...)
+	}
+	for _, sampler := range envelope.gauges {
+		ms = append(ms, sampler.Flush()...)
+	}
+	for _, sampler := range envelope.histograms {
+		ms = append(ms, sampler.Flush(time.Second, s.percentiles, s.aggregates, true)...)
+	}
+	for _, sampler := range envelope.mixedHistograms {
+		ms = append(ms, sampler.Flush(s.percentiles, s.aggregates, envelope.mixedHosts)...)
+	}
+
+	if len(ms) == 0 {
+		return
+	}
+
+	tags := map[string]string{"part": "post"}
+	for _, sinkInstance := range s.sinks {
+		// TODO(clin): Add back ms once we finalize the ms client pull request.
```

Comment:
What's `ms`?

Reply:
it's supposed to say metrics which are back in, will remove comment